backend/main/src/Services/DeleteUserService/DeleteUserService.ts

```diff
@@ -0,0 +1,165 @@
+import {
+	BadRequestStatus,
+	Controller,
+	InternalServerErrorStatus,
+} from "src/Shared";
+import { IGroupRepository } from "src/Group/Domain/IGroupRepository";
+import { FindGroupsInformationQuery } from "src/Group/Query/FindGroupsInformation/FindGroupsInformatoinQuery";
+import { UserId } from "src/User";
+import { Group, GroupId } from "src/Group/Domain/Entities/Group";
+import { IUserRepository } from "src/User/Domain/IUserRepository";
+import {
+	AdminDeleteUserCommand,
+	AdminDeleteUserCommandInput,
+	CognitoIdentityProviderClient,
+	ListUsersCommand,
+	ListUsersCommandInput,
+} from "@aws-sdk/client-cognito-identity-provider";
+import { COGNITO_USER_POOL_ID } from "src/Settings/Setting";
+import { Err, Ok, Result } from "result-ts-type";
+
+interface IDeleteUserProps {
+	/** requested by the user */
+	userId: string;
+	/** from the token */
+	targetUserId: string;
+}
+
+/**
+ * Delete a user.
+ * The groups and containers that are possessed by only this user.
```

Comment:
To make it clear, it seems better to be explicit that the groups and containers are deleted.
```suggestion
 * The groups and containers are deleted when they are possessed by only this user.
```